src/utils/validateTrimmedString.ts

```diff
@@ -1,10 +1,41 @@
-// makes sure the string passed into the function isn't an empty string
-export function validateTrimmedString(input: string) {
-	const trimmedInput = input.trim();
+import { ListItem } from "../api";
 
+// Validates the item name input for a shopping list
+export function validateItemName(
+	input: string,
+	existingItems: ListItem[],
+): string | null {
+	const trimmedInput = input.trim(); //removes leading and trailing whitespaces
+
+	// Condition 1: Check if the input is empty
 	if (trimmedInput.length === 0) {
-		return null;
+		return "Item cannot be empty";
+	}
+
+	//Remove punctuation marks and normalize input
+	const punctuationRegex = /[^\p{L}]/gu;
+
+	const normalizedInputName = trimmedInput
+		.replace(punctuationRegex, "")
+		.toLowerCase();
+
+	//Create a list of  normalized existing item names
+	const normalizedExistingItemNames = existingItems.map((existingItem) => {
+		return existingItem.name.replace(punctuationRegex, "").toLowerCase();
+	});
+
+	// Condition 2: Check if the normalized input matches any existing item
+	const isDuplicateItem = (normalizedInputName: string): boolean => {
+		return normalizedExistingItemNames.some(
+			(item) => item === normalizedInputName,
+		);
+	};
+
+	//return error if the item already exists
+	if (isDuplicateItem(normalizedInputName)) {
+		return ` ${normalizedInputName} already exists in the list`;
```

Comment:
Nice job specifying to the user the exact error when duplicate entries are submitted!